src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -45,36 +43,19 @@
   }
 
   {{/zcl_events}}
-  {{#chip_server_cluster_attributes}}
-  {{#unless isGlobalAttribute}}
-  {{! ensure indent }}{{#unless isWritableAttribute~}}
-    readonly {{!marker to place a space even with whitespace removal~}}
-  {{~/unless~}}
-  {{~!TODO: write only attributes should also be supported~}}
-  {{~#unless isReportableAttribute~}}
-    nosubscribe {{!marker to place a space even with whitespace removal~}}
-  {{~/unless~}}
-  {{~!Removes spaces~}} attribute {{!ensure space}}
-    {{~#chip_access_elements entity="attribute"~}}
-      {{~#first~}}access({{~/first~}}
-      {{~#not_first~}}, {{/not_first~}}
-        {{operation}}: {{role}} 
-      {{~#last}}) {{/last~}}
-    {{~/chip_access_elements~}}
-    {{~#if isNullable~}}
-      nullable {{!marker to place a space even with whitespace removal~}}
-    {{~/if~}} {{type}}
-    {{~#unless isArray~}}
-      {{~#if (isString type)~}}
-        {{~#if maxLength~}} 
-          <{{maxLength}}> 
-        {{~/if~}}
+  {{#if (is_client side)}}
+    {{#zcl_attributes_server}}
+      {{~#if clusterRef}}{{!Using `ClusterRef` to detect global attributes~}}
+        {{~>idl_cluster_attribute~}}
       {{~/if~}}
-    {{/unless}} {{asLowerCamelCase name~}}
-    {{~#if isArray~}} [] {{~/if}} = {{code~}}
-    ;
-  {{/unless~}}
-  {{/chip_server_cluster_attributes~}}
+    {{/zcl_attributes_server~}}
+  {{else}}
+    {{#chip_server_cluster_attributes}}
+      {{~#unless isGlobalAttribute~}}
```

Comment:
Why do you need to switch to `chip_server_cluster_attributes` here?
Are you looking for `if_basic_attribute` by any chance?

Reply:
chip_server_cluster_attributes was used before already before. I only switched the client-side currently. This piece of code is unchanged from what was before (except I moved the attribute generation into a common file to re-use it)